### PR TITLE
Opaque pointer: Update createDwordAtAddr

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**
Function will accept base type and address space and later inside createDwordAtAddr the pointer will be created

### CODE REVIEW

File: llpc/lower/llpcSpirvLowerRayTracingIntrinsics.cpp [hunk 1]
```diff
@@ -134,11 +134,15 @@ bool SpirvLowerRayTracingIntrinsics::processIntrinsicsFunction(Function *func) {
 // Create AmdExtD3DShaderIntrinsics_LoadDwordAtAddr, LoadDwordAtAddrx2, LoadDwordAtAddrx4,
 //
 // @param func : Function to create
-// @param loadPtrTy : Pointer type of the load value
-void SpirvLowerRayTracingIntrinsics::createLoadDwordAtAddr(Function *func, Type *loadPtrTy) {
+// @param loadTy : Base type of the load value
+// @param addressSpace : Address space of the memory to load the value from
+void SpirvLowerRayTracingIntrinsics::createLoadDwordAtAddr(Function *func, Type *loadTy,
+                                                           SPIRAddressSpace addressSpace) {
```
Comment (Flakebi) [inline llpc/lower/llpcSpirvLowerRayTracingIntrinsics.cpp:140]: I think we can omit the address space argument, the function always creates a pointer to the global addrspace.

Comment (mariusz-sikora-at-amd) [reply llpc/lower/llpcSpirvLowerRayTracingIntrinsics.cpp:140]: > I think we can omit the address space argument, the function always creates a pointer to the global addrspace.

Right now yes.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 008d8330b1ec7e759b1fadab32c1e2afa990c34d
<details><summary>CTS tests (Failed: 0/286514)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72648 (49.9%)</li>
<li>Failed: 0/72648 (0.0%)</li>
<li>Not Supported: 36426/72648 (50.1%)</li><li>Warnings: 0/72648 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40420/72695 (55.6%)</li>
<li>Failed: 0/72695 (0.0%)</li>
<li>Not Supported: 32275/72695 (44.4%)</li><li>Warnings: 0/72695 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72702 (51.1%)</li>
<li>Failed: 0/72702 (0.0%)</li>
<li>Not Supported: 35567/72702 (48.9%)</li><li>Warnings: 0/72702 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68469 (50.9%)</li>
<li>Failed: 0/68469 (0.0%)</li>
<li>Not Supported: 33635/68469 (49.1%)</li><li>Warnings: 0/68469 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]

**Comment by mariusz-sikora-at-amd** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 18bbc420711de19b7c6e1b6bc9f6a5c605d5f139
<details><summary>CTS tests (Failed: 1/286530)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72652 (49.9%)</li>
<li>Failed: 0/72652 (0.0%)</li>
<li>Not Supported: 36430/72652 (50.1%)</li><li>Warnings: 0/72652 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40418/72699 (55.6%)</li>
<li><details><summary>Failed: 1/72699 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.ray_tracing_pipeline.barrier.simg.specific_barrier.from_miss_to_xfer
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 32280/72699 (44.4%)</li><li>Warnings: 0/72699 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72706 (51.1%)</li>
<li>Failed: 0/72706 (0.0%)</li>
<li>Not Supported: 35571/72706 (48.9%)</li><li>Warnings: 0/72706 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68473 (50.9%)</li>
<li>Failed: 0/68473 (0.0%)</li>
<li>Not Supported: 33639/68473 (49.1%)</li><li>Warnings: 0/68473 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 18bbc420711de19b7c6e1b6bc9f6a5c605d5f139
<details><summary>CTS tests (Failed: 0/286530)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72652 (49.9%)</li>
<li>Failed: 0/72652 (0.0%)</li>
<li>Not Supported: 36430/72652 (50.1%)</li><li>Warnings: 0/72652 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40419/72699 (55.6%)</li>
<li>Failed: 0/72699 (0.0%)</li>
<li>Not Supported: 32280/72699 (44.4%)</li><li>Warnings: 0/72699 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72706 (51.1%)</li>
<li>Failed: 0/72706 (0.0%)</li>
<li>Not Supported: 35571/72706 (48.9%)</li><li>Warnings: 0/72706 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68473 (50.9%)</li>
<li>Failed: 0/68473 (0.0%)</li>
<li>Not Supported: 33639/68473 (49.1%)</li><li>Warnings: 0/68473 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**